fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneAutoCompleteQueryClause.java

```diff
@@ -309,16 +311,14 @@ public static Query buildQueryForPhraseMatching(@Nonnull QueryParser parser,
         // Construct a query that is essentially:
         //  - in any field,
         //  - the phrase must occur (with possibly the last token in the phrase as a prefix)
+        List<String> tokens = new ArrayList<>();
         String phrase = null;
-        String prefix = null;
-        final int prefixBegin = searchKey.lastIndexOf(" ");
-
-        if (prefixBegin > -1) {
-            phrase = searchKey.substring(0, prefixBegin);
-        }
+        String prefix = getQueryTokens(new AutoCompleteAnalyzer(), searchKey, tokens);
 
-        if (prefixBegin < searchKey.length() - 1) {
-            prefix = searchKey.substring(prefixBegin + 1);
+        if (!tokens.isEmpty()) {
+            final String lastToken = tokens.get(tokens.size() - 1);
+            final int phraseEnd = searchKey.lastIndexOf(lastToken) + lastToken.length();
+            phrase = searchKey.substring(0, phraseEnd);
```

Comment:
Yes, essentially. Or strip off non-tokens between the last token before the prefix and the prefix if there is one.

Reply:
"non-tokens" meaning that `John likes the` would be converted to `John likes`, correct?

Reply:
the `AutoCompleteAnalyzer` does not filter out stopwords, so if the part before the prefix is `John likes the`, it would still be `John likes the`. The `the` would then be treated as a stopword if the index analyzer used by the Lucene query parser passed into `buildQueryForPhraseMatching` considers it a stopword.

Reply:
In this case it would be punctuations and what the `UAX29URLEmailTokenizer` considers word breaks. `John likes the, ;` would become `John likes the`